includes/publisher/js/src/components/Field.jsx

```diff
@@ -87,6 +88,16 @@ export default function Field(props) {
 			if (field?.required && event.target.value.trim() === "") {
 				error = defaultError;
 			}
+
+			if (
+				field?.allowedDomains &&
+				event.target.validity.patternMismatch
+			) {
+				error = __(
+					"Email must include domain or subdomain criteria.",
+					"atlas-content-modeler"
+				);
+			}
```

Comment:
Could this message be clearer? Perhaps:

```suggestion
				error = __(
					"Email must end with an allowed domain.",
					"atlas-content-modeler"
				);
			}
```

(Just mirroring the "allowed domains" that you use above the field, which is great, and perhaps more obvious than “…domain criteria”.)